internal/inventory/asset.go

```diff
@@ -0,0 +1,169 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package inventory
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+)
+
+type assetCategory string
+type assetSubCategory string
+type assetType string
+type assetSubType string
+type assetCloudProvider string
+
+const (
+	CategoryInfrastructure assetCategory = "infrastructure"
+
+	SubCategoryCompute assetSubCategory = "compute"
+
+	TypeVirtualMachine assetType = "virtual-machine"
+
+	SubTypeEC2 assetSubType = "ec2"
+
+	AwsCloudProvider assetCloudProvider = "aws"
+)
```

Comment:
I know this is just a stub, but it would be neat to break down the `const` block. There will be hundreds of Categories, SubTypes and Types by the end of implementation. How about an `enum`? We can split those by cloud provider, so its easier to know `SubCategoryCompute` goes with `AwsCloudProvider` and not the others. 
Just something to consider.

Reply:
I agree.

For now I'll just split. Once we have more cloud providers we can think of a more elaborated way to organise stuff